chainindex/gc.go

```diff
@@ -0,0 +1,92 @@
+package chainindex
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/filecoin-project/lotus/chain/actors/builtin"
+)
+
+var (
+	log             = logging.Logger("chainindex")
+	cleanupInterval = time.Duration(4) * time.Hour
+)
+
+func (si *SqliteIndexer) gcLoop() {
+	defer si.wg.Done()
+
+	// Initial cleanup before entering the loop
+	si.gc(si.ctx)
+
+	cleanupTicker := time.NewTicker(cleanupInterval)
+	defer cleanupTicker.Stop()
+
+	for si.ctx.Err() == nil {
```

Comment:
probably worth doing a `si.closed` check in this loop

Reply:
Done.